azdev/operations/help/refdoc/extension_docs/helpgen.py

```diff
@@ -0,0 +1,91 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azdev.operations.help.refdoc.common.directives import AbstractHelpGenDirective
+from azdev.operations.help.refdoc.common.directives import setup_common_directives
+
+from knack.log import get_logger
+from knack.help import GroupHelpFile
+
+from azure.cli.core.file_util import _store_parsers, _is_group
+from azure.cli.core.commands import ExtensionCommandSource
+from azure.cli.core._help import CliCommandHelpFile
+
+logger = get_logger(__name__)
+
+class ExtensionHelpGenDirective(AbstractHelpGenDirective):
+    """
+        CLI Extensions Sphinx Directive
+        Ensure that sphinx output is generated from only extension help files
+    """
+
+    def _get_help_files(self, az_cli):
+        return get_extension_help_files(az_cli)
+
+    def _load_doc_source_map(self):
+        # no doc source map for extensions
+        pass
+
+    def _yield_doc_source(self, doc_source_map, help_file):
+        # no doc source map for extensions
+        pass
+
+# TODO: move this into core
```

Comment:
We should really add logic like this into the core, as it is similar to get_all_help